src/sdk/nsfs_glacier_backend/backend.js

```diff
@@ -153,7 +169,22 @@ class GlacierBackend {
      * @returns {nb.RestoreStatus | undefined}
      */
     static get_restore_status(xattr, now, file_path) {
-        if (xattr[GlacierBackend.STORAGE_CLASS_XATTR] !== s3_utils.STORAGE_CLASS_GLACIER) return;
+		if (GlacierBackend.storage_class_from_xattr(xattr, config.NSFS_GLACIER_USE_DMAPI) !== s3_utils.STORAGE_CLASS_GLACIER) {
+			return;
+		}
+
+        if (config.NSFS_GLACIER_USE_DMAPI) {
+            if (xattr[GlacierBackend.GPFS_DMAPI_XATTR_TAPE_PREMIG]) {
+                const expiry = new Date();
```

Comment:
this expiry might deserve a distinct name, like
```suggestion
                const premig_expiry = new Date();
```